### PR TITLE
Create submit button util

### PR DESCRIPTION
**Pull request by jkerr321**
## Feature Description
In order to allow text on the submit button to be dynamically updated 🐿 v2.12.3

## Link to Ticket / Card:
https://trello.com/c/G4VNkcM8
https://trello.com/c/zx00FXFh

Linked to https://github.com/Financial-Times/next-subscribe/pull/233

## Screenshots:
image
image

## Has the necessary documentation been created / updated?
- [x] Not required for this ticket

### CODE REVIEW

**Comment by liamkeaton** [comment]
@jkerr321 could you update the README.md with this new functionality, then you can gain the 👍 of approval

**Comment by jkerr321** [comment]
@liamkeaton Ah yes, how bold of me to mark this as `documentation not required` without even looking at the README first and seeing that all the utils are, in fact, beautifully documented 😬

File: utils/submit.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+		if (!this.$submit) {
+			throw new Error('Please include the submit button partial on the page');
+		}
+	}
+
+	/**
+	 * Update the button text
+	 * @param {String}
+	 */
+	updateText (newText) {
```
Comment (richardLuko) [inline utils/submit.js]: should it be just `text` ?

Comment (jkerr321) [reply utils/submit.js]: There will be default text on the button already (in most cases), hence `newText` was to kinda indicate that it's replacement/new text compared with what's there already

Comment (liamkeaton) [reply utils/submit.js]: Think the method name implies that though?

File: tests/utils/submit.spec.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+			expect(() => {
+				new Submit();
+			}).to.throw();
+		});
+
+		it('should throw an error if button not present', () => {
+			expect(() => {
+				documentStub.querySelector.returns(false);
+				new Submit(documentStub);
+			}).to.throw();
```
Comment (umbobabo) [inline tests/utils/submit.spec.js]: Wouldn't be better test the specific message `.to.throw('Please supply a DOM element')`? I see the risk that potentially you could intercept errors from others part of the code.

File: tests/utils/submit.spec.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+	afterEach(() => {
+		sandbox.restore();
+	});
+
+	describe('constructor', () => {
+		it('should throw an error if nothing passed', () => {
+			expect(() => {
+				new Submit();
+			}).to.throw();
```
Comment (umbobabo) [inline tests/utils/submit.spec.js]: See comment below.

File: tests/utils/submit.spec.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+			documentStub.querySelector.returns(elementStub);
+			submit = new Submit(documentStub);
+
+		});
+
+		describe('updateText', () => {
+			it('should throw if no string given', () => {
+				expect(() => {
+					submit.updateText();
+				}).to.throw();
```
Comment (umbobabo) [inline tests/utils/submit.spec.js]: Same here.

File: utils/submit.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+ */
+
+const SUBMIT_BUTTON_CLASS = '.ncf__button--submit';
+
+class Submit {
+	/**
+	 * Initalise the submit utility
+	 * @param {Element} element Usually the window.document
+	 * @throws If the document not passed
+	 * @throws When the submit element not found
```
Comment (umbobabo) [inline utils/submit.js]: Maybe you could add the `@return`?

File: utils/submit.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+ *
+ * // Update the button text
+ * submit.updateText('Pay with Apple Pay');
+ */
+
+const SUBMIT_BUTTON_CLASS = '.ncf__button--submit';
+
+class Submit {
+	/**
+	 * Initalise the submit utility
```
Comment (umbobabo) [inline utils/submit.js]: I think there is a typo in this sentence.

**Pull request merged**